packages/client/src/App.tsx

```diff
@@ -1,31 +1,35 @@
-import { useComponentValue } from "@latticexyz/react";
-import { SyncStep } from "@latticexyz/store-sync";
-import { singletonEntity } from "@latticexyz/store-sync/recs";
-import { useMUD } from "./MUDContext";
-import { GameBoard } from "./GameBoard";
+import { useState } from "react";
+import { ToastContainer } from "react-toastify";
+import { LoadingWrapper } from "./LoadingWrapper";
+import { createBurnerWalletClient } from "./mud/account";
+import { useDevTools } from "./mud/devTools";
+import { useFaucet } from "./mud/faucet";
+import { useSetup } from "./mud/setup";
+import { WalletClientWithAccount } from "./mud/types";
+import { MUDProvider } from "./MUDContext";
+
+const walletClient = createBurnerWalletClient()
 
 export const App = () => {
-  const {
-    components: { SyncProgress },
-  } = useMUD();
+  const [wallet, setWallet] = useState<WalletClientWithAccount | undefined>();
+  const mud = useSetup(wallet);
+  useFaucet(wallet?.account.address);
+  useDevTools(mud);
 
-  const loadingState = useComponentValue(SyncProgress, singletonEntity, {
-    step: SyncStep.INITIALIZE,
-    message: "Connecting",
-    percentage: 0,
-    latestBlockNumber: 0n,
-    lastBlockNumberProcessed: 0n,
-  });
+  const connect = () => setWallet(walletClient);
+  const disconnect = () => setWallet(undefined);
```

Comment:
Obviously silly here since we have the burner from the get-go, but any login/wallet system can inject its own wallet client.